platform/darwin/test/MGLOfflineStorageTests.mm

```diff
@@ -264,4 +264,107 @@ - (void)testResourceTransform {
     [os setDelegate:nil];
 }
 
+- (void)testAddingFileContent {
+    
+    NSURL *styleURL = [MGLStyle streetsStyleURLWithVersion:MGLStyleDefaultVersion];
+    
+    // Barcelona.
+    MGLCoordinateBounds bounds = {
+        { .latitude = 41.2724, .longitude = 2.0407 },
+        { .latitude = 41.4664, .longitude = 2.2680 },
+    };
+    double zoomLevel = 20;
+    MGLTilePyramidOfflineRegion *region = [[MGLTilePyramidOfflineRegion alloc] initWithStyleURL:styleURL bounds:bounds fromZoomLevel:zoomLevel toZoomLevel:zoomLevel];
+    
+    NSString *nameKey = @"Name";
+    NSString *name = @"Barcelona";
+    
+    NSData *context = [NSKeyedArchiver archivedDataWithRootObject:@{
+                                                                    nameKey: name,
+                                                                    }];
+    
+
+    XCTestExpectation *additionCompletionHandlerExpectation = [self expectationWithDescription:@"add pack completion handler"];
+    [[MGLOfflineStorage sharedOfflineStorage] addPackForRegion:region withContext:context completionHandler:^(MGLOfflinePack * _Nullable completionHandlerPack, NSError * _Nullable error) {
+        XCTAssertNotNil(completionHandlerPack, @"Added pack should exist.");
+        XCTAssertEqual(completionHandlerPack.state, MGLOfflinePackStateInactive, @"New pack should initially have inactive state.");
+        [additionCompletionHandlerExpectation fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:2 handler:nil];
+    
+    // Valid database
+    {
+        NSURL *resourceURL = [NSURL fileURLWithPath:[[NSBundle bundleForClass:[self class]] pathForResource:@"barcelona" ofType:@"db"]];
+        NSArray *documentPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+        NSString *documentDir = [documentPaths objectAtIndex:0];
+        NSString *filePath = [documentDir stringByAppendingPathComponent:@"barcelona.db"];
+        NSFileManager *fileManager = [NSFileManager defaultManager];
+        
+        BOOL exists = [fileManager fileExistsAtPath:filePath];
+        if (exists) {
+            [fileManager removeItemAtPath:filePath error:nil];
+        }
+        
+        NSError *error;
+        [fileManager moveItemAtURL:resourceURL toURL:[NSURL fileURLWithPath:filePath] error:&error];
+        
+        XCTestExpectation *fileAdditionCompletionHandlerExpectation = [self expectationWithDescription:@"add database content completion handler"];
+        MGLOfflineStorage *os = [MGLOfflineStorage sharedOfflineStorage];
+        [os addContentsOfFile:filePath withCompletionHandler:^(NSError * _Nullable error) {
+            XCTAssertNil(error, @"Adding contents to a file should not return an error.");
```

Comment:
Also assert that new packs are included in `MGLOfflineStorage.packs` by this point.

/ref https://github.com/mapbox/mapbox-gl-native/pull/12791#discussion_r215794666